lib/nnc/cmd/blas/mps/ccv_nnc_gemm_mps.m

```diff
@@ -352,7 +352,148 @@ static int _ccv_nnc_gemm_forw(const ccv_nnc_cmd_t cmd, const ccv_nnc_hint_t hint
 }
 
 static int _ccv_nnc_gemm_back(const ccv_nnc_cmd_t cmd, const ccv_nnc_hint_t hint, const int flags, ccv_nnc_tensor_t* const* const inputs, const int input_size, ccv_nnc_tensor_t* const* const outputs, const int output_size, ccv_nnc_stream_context_t* const stream_context)
-{
+{	
+	// inputs: gradient g, forw prop input a, [w]
+	// outputs: output gradient h, weight updates dw, bias updates bias
+	assert(input_size >= 2 && output_size >= 2);
+	const ccv_nnc_tensor_view_t* g = (const ccv_nnc_tensor_view_t*)inputs[0];
+	ccv_nnc_tensor_view_t* dw = output_size > 1 ? (ccv_nnc_tensor_view_t*)outputs[1] : 0;
+	ccv_nnc_tensor_view_t* bias = output_size > 2 ? (ccv_nnc_tensor_view_t*)outputs[2] : 0;
+
+	const ccv_nnc_tensor_view_t* a = dw ? (const ccv_nnc_tensor_view_t*)inputs[1] : 0;
+	ccv_nnc_tensor_view_t* h = (ccv_nnc_tensor_view_t*)outputs[0];
+	const ccv_nnc_tensor_view_t* w = h ? (const ccv_nnc_tensor_view_t*)inputs[2] : 0;
+
+	assert(!bias || (bias->info.dim[1] == 0 || bias->info.dim[2] == 0 || bias->info.dim[3] == 0)); // It is a 2-d or 3-d array.
```

Comment:
This comment is strange, this asserts bias is always a 1-d array.

Reply:
will update it to 1-d in comments...
btw.. all other gemm back said 2-d or 3-d as well... 
 https://github.com/search?q=repo%3Aliuliu%2Fccv+%22%2F%2F+It+is+a+2-d+or+3-d+array.%22&type=code

Reply:
Oh, I mis-read, you are right. The original comment is correct.